lesson16/index.js

```diff
@@ -0,0 +1,158 @@
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+  allData.forEach((content, index) => {
+    if (content.isActive) {
+      primaryImageWrapper.classList.add("news_Img");
+      const img = document.createElement("img");
+      img.src = content.image.img;
+      primaryImageWrapper.appendChild(img);
+
+      renderNewsList(content, lists);
+      const tabItem = document.querySelector(
+        ".news_Tabs .tab_item:nth-child(1)"
+      );
+      tabItem.classList.add("is-active");
+    }
+  });
+
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+    const tabsItems = Array.from(newsTabs.querySelectorAll("li"));
+
+    // タブのindexを取得し、スタイル変更
+    tabsItems.forEach((tab) => {
+      tab.addEventListener("click", (e) => {
+        const activeIndex = tabsItems.filter((tab) => {
+          return tab === e.target;
+        });
+        const inactiveIndex = tabsItems.filter((tab) => {
+          return tab !== e.target;
+        });
+
+        activeIndex[0].classList.add("is-active");
+        inactiveIndex.forEach((element) => {
+          element.classList.remove("is-active");
+        });
+      });
+    });
+
+    if (target.textContent === "news") {
+      console.log("news");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[0].image.img;
+      img.alt = allData[0].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[0], lists);
+    } else if (event.target.textContent === "economy") {
+      console.log("economy");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[1].image.img;
+      img.alt = allData[1].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[1], lists);
+    } else if (event.target.textContent === "sports") {
+      console.log("sports");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[2].image.img;
+      img.alt = allData[2].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[2], lists);
+    }
+  });
+};
+
+const renderNewsList = (content, lists) => {
```

Comment:
```suggestion
const renderNewsList = ({article}) => {
```
IMO
You can try using [Destructuring assignment](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

Also, the `lists` received as an argument and the `newsList` defined in the function point to the same element, I don't think the `lists` argument is necessary.

Reply:
Q.
This part is IMO, is it intentional that it is not supported?

Reply:
I forgot to fix this😱
Fixed to use split assignment!